networkx/algorithms/richclub.py

```diff
@@ -108,6 +110,9 @@ def _compute_rc(G):
     # side of the list, which would have a linear time cost.
     edge_degrees = sorted((sorted(map(G.degree, e)) for e in G.edges()), reverse=True)
     ek = G.number_of_edges()
+    if ek == 0:
+        return {}
```

Comment:
you can raise `nx.NetworkXNotImplemented("G does not have any edges.")` exception here. It would be more helpful to the user.

Reply:
{} is a valid output for an edgeless graph, right? Like I think it's pestering the user even for valid inputs. I can write in the docstring, or a comment above the if statement about this case though, should I do that?